### PR TITLE
Feature/ent transaction

### PR DESCRIPTION
**Pull request by miraj90van**
WIP - transaction

### CODE REVIEW

File: ent/schema/post.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// Fields of the Post.
+func (Post) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("title").NotEmpty(),
+		field.String("content").NotEmpty(),
+		field.String("slug").NotEmpty(),
+		field.Int("status").NonNegative(),
+		field.Time("created_at").Default(time.Now),
+		field.Time("updated_at").Optional().Nillable(),
+		field.Time("deleted_at").Optional().Nillable(),
```
Comment (adityasatrio) [inline ent/schema/post.go:23]:
```
// Fields of the System_parameter.
func (System_parameter) Fields() []ent.Field {
	schema := []ent.Field{
		field.String("key").NotEmpty().Unique(),
		field.String("value").NotEmpty(),
	}

	return helper.InitBaseSchema(schema)
}
```

pakai `InitBaseSchema` supaya semua nya sama

File: internal/applications/post/controller/post_controller.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	if err != nil {
+		return err
+	}
+
+	err = ctx.Validate(request)
+	if err != nil {
+		return err
+	}
+
+	created, err := c.service.Create(ctx.Request().Context(), *request)
```
Comment (adityasatrio) [inline internal/applications/post/controller/post_controller.go:31]:
```
	created, err := c.service.Create(ctx.Request().Context(), *request)
```

`request` kenapa pake pointer ya, menurut gw cukup passing by value nya aja krn ga bakal di modify lagi

File: internal/applications/post/dto/post_dto.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+type (
+	PostRequest struct {
+		Title   string `json:"title" validate:"required"`
+		Content string `json:"content" validate:"required"`
+		Slug    string `json:"slug" validate:"required"`
+		Status  int    `json:"status"`
+	}
+
+	Post struct {
```
Comment (adityasatrio) [inline internal/applications/post/dto/post_dto.go:11]:
```suggestion
	PostResponse struct {
```

**Review by adityasatrio — CHANGES_REQUESTED** [review]
request minor changes

**Comment by miraj90van** [comment]
close pr, use another pr

**Pull request closed without merging**